src/storage/mvcc/reader/scanner/forward.rs

```diff
@@ -558,14 +560,59 @@ fn scan_latest_handle_lock<S: Snapshot, T>(
 /// (or locks' `start_ts`s) in range (`from_ts`, `cfg.ts`].
 pub struct DeltaEntryPolicy {
     from_ts: TimeStamp,
+    extra_op: ExtraOp,
 }
 
 impl DeltaEntryPolicy {
-    pub fn new(from_ts: TimeStamp) -> Self {
-        Self { from_ts }
+    pub fn new(from_ts: TimeStamp, extra_op: ExtraOp) -> Self {
+        Self { from_ts, extra_op }
     }
 }
 
+fn seek_for_valid_value<S: Snapshot>(
+    cursors: &mut Cursors<S>,
+    user_key: &Key,
+    after_ts: TimeStamp,
+    statistics: &mut Statistics,
+) -> Result<Option<Value>> {
+    let mut value = None;
+    while cursors.write.valid()?
+        && Key::is_user_key_eq(
+            cursors.write.key(&mut statistics.write),
+            user_key.as_encoded(),
+        )
+    {
+        assert_ge!(
+            after_ts,
+            Key::decode_ts_from(cursors.write.key(&mut statistics.write))?
+        );
+        let write_ref = WriteRef::parse(cursors.write.value(&mut statistics.write))?;
+        match write_ref.write_type {
+            WriteType::Put => {
+                // Read the value of the write record.
+                value = if let Some(v) = write_ref.short_value.map(|v| v.to_vec()) {
+                    Some(v)
+                } else {
+                    let default_cursor = cursors.default.as_mut().unwrap();
+                    Some(super::near_load_data_by_write(
+                        default_cursor,
+                        user_key,
+                        write_ref.start_ts,
+                        statistics,
+                    )?)
+                };
+                break;
+            }
+            WriteType::Delete => break,
+            WriteType::Lock | WriteType::Rollback => {
```

Comment:
It seems in this PR we will skip some `Lock` for reading the old value. However, the current implementation outputs `Lock`. Will this behavior change affect CDC? @MyonKeminta